overlord/devicestate/firstboot.go

```diff
@@ -399,6 +416,13 @@ func (m *DeviceManager) populateStateFromSeedImpl(tm timings.Measurer) ([]*state
 		return nil, err
 	} else if trackVss != nil {
 		trackVss.WaitAll(ts)
+
+		// if this validation fails, we want to undo the tasks in these lanes.
+		// TODO: currently, undoing seeding the essential set of snaps does not
+		// work properly
+		trackVss.JoinLane(essentialLane)
+		trackVss.JoinLane(nonEssentialLane)
```

Comment:
wouldn't be enough to just add this to nonEssentialLane,  even if this fails we still don't want to remove the essential snaps, no?